File: DerekTromProgram1/TromDHW1.py
'''
Derek Trom
derek.trom@und.edu
HW 1 CSCI 365
Random Num List Program
'''
import random

def getMin(array):
    minimum = array[0]
    for i in range(0,20):
        if array[i] < minimum:
            minimum = array[i]
    return minimum
def fillArray():
    array = []
    for i in range(0, 20):
        array.append(random.randint(0,100))
    return array
def getMax(array):
    maximum = array[0]
    for i in range(0,20):
        if array[i] > maximum:
            maximum = array[i]
    return maximum
def printArray(array):
    count = 0
    for i in array:
        count += 1
        print("%3d %3d" % (count, i))
def findAverage(array):
    summation = 0
    count = 0
    for i in array:
        count+= 1
        summation += i
    return summation/count
    

if __name__ == '__main__':
    
    randomArray = fillArray()
    minimum = getMin(randomArray)
    print("Minimum: %d" % minimum)
    maximum = getMax(randomArray)    
    print("Maximum: %d" % maximum)
    averageOfArray = findAverage(randomArray)
    print("Average: %.2f" % averageOfArray)
    print("Values")
    printArray(randomArray)
                

File: DerekTromProgram1/TromDHW1.java
/*
  Derek Trom
  derek.trom@und.edu
  HW1 CSCI 365
 */
import java.util.Random;
public class TromDHW1{
    public static int generateRandomInt(int upperRange){
        Random random = new Random();
        return random.nextInt(upperRange);
    }
    public static int[] fillArray(int []array) {
        for (int i = 0, n = array.length; i < n; i++) {
            array[i] = generateRandomInt(100);
        }
        return array;
    }
    public static void printArray(int[] array){
        int count = 0;
        for (int i = 0, n = array.length; i < n; i++) {
            count++;
            System.out.printf("%3d %3d\n" ,count, array[i]);
        }

    }
    public static int maxOfArray(int[] array){
        int max = array[0];
        for (int i = 0, n = array.length; i < n; i++){
            if (array[i] > max){
                max = array[i];
            }
        }
        return max;
    }
    public static int minOfArray(int[] array){
        int min = array[0];
        for (int i = 0, n = array.length; i < n; i++){
            if (array[i] < min){
                min = array[i];
            }
        }
        return min;
    }
    public static double avgOfArray(int[] array){
        int sum = 0;
        int count = 0;
        for (int i = 0, n = array.length; i < n; i++){
            count++;
            sum += array[i];
        }
        double average = sum/count;

        return average;
    }


    public static void main(String[] args){

        int arrayOfNums[] = new int[20];
        arrayOfNums = fillArray(arrayOfNums);
        int max = maxOfArray(arrayOfNums);
        System.out.println("Max: " + max);
        int min = minOfArray(arrayOfNums);
        System.out.println("Min: "+ min);
        double average = avgOfArray(arrayOfNums);
        System.out.printf("Average = %.2f\n", average);
        System.out.print("Values\n");
        printArray(arrayOfNums);


    }
}